### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by appiersign**
In this pull request we:
- removed comments in the main.rb file
- added methods to create new board
- added player to create new players
- added checks or cell availability before registering players move
- added validation for player names
- added checks for tie and winner

### CODE REVIEW

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
The project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team!  
### You did a great job so far :+1: 
Still, there are some changes required before this Pull request can be approved:
* [ ] The application should inform a player about invalid input and asks for their move again. Right now it just asks the player to select a number to play. 
* [ ] All classes should be kept in `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)) Right now you have `Game` class in `main.rb` file. 
* [ ] The player is allowed to have an empty name in your game. It would be great if you add some validation to prevent it.
* [ ]  Right now it's hard to play a game because it's displayed in one line, try to make it look more user friendly. Also, the game is displaying values in between boards, that should be removed:
image
### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: bin/main.rb
Comment (majovanilla) [inline bin/main.rb]: Like the previous TSE explained, all classes should be on the lib folder. To be more specific, all logic should be inside the main classes you have. In this case, the player_move method should be inside the player class. Please move it to the lib folder.

File: bin/main.rb
Comment (majovanilla) [inline bin/main.rb]: This method should also be inside the board class on the lib folder.

File: bin/main.rb
Comment (majovanilla) [inline bin/main.rb:6]: You shouldn't keep the players' array here. You can create a method inside the player class that stores all this information and you can call it when the players input their names.

File: bin/main.rb
Comment (majovanilla) [inline bin/main.rb:20]: You could move this into the method I mentioned in the first comment so that when the user inputs the name, you can push it into the players' array you created. Tip: the method would take the user's input as an argument, so you don't have to use puts or chomp on the class.

File: lib/board.rb
Comment (majovanilla) [inline lib/board.rb]: Your board still has some minor details that you could improve. For example, the width of the cells is one with the empty board, but when the players start checking them, the width changes and the board loses the proportion and it's difficult to see the columns.
Screenshot from 2020-04-22 20-57-03

File: lib/board.rb
Comment (majovanilla) [inline lib/board.rb:120]: You are not displaying the final move on the board. When a player wins, the game declares the winner and it never shows the winning board. You should show it before declaring the winner.

Your game is not ending when there is a draw and it allows the players to keep selecting cells even when they are no available anymore.

File: lib/board.rb
Comment (majovanilla) [inline lib/board.rb]: Your method is not working. I can select taken cells with no warning and I can also input a larger or smaller number with no warning. The only thing that it's doing it's not changing turns, but that is not enough, you should tell the player that the move was not valid.

File: lib/board.rb
Comment (majovanilla) [inline lib/board.rb:84]: **Optional** an easier way to check for the winner is to have all possible combinations and check the array against them, using the player's character as an argument.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hi team!! 

## You did a *great* job on this project! 💜 

You are almost done!! It's going to look amazing. Just a few other requirements to make it work as it should. Please check the commented code below.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @appiersign and @aulbytj :lollipop: :balloon: 

### Thank you for doing most of the changes requested by the previous reviewers :soccer: :medal_sports: , I always like when the game allows custom names for the players it makes it more human friendly :video_game: I think you've done enough with this project so is a yes for me you may move on congratulations :tada: :tada: :tada: 

[OPTIONAL]

- [ ] I agree with @majovanilla this section of your code should be achieved with methods in your players class, it would make your code more readable to other programmers so I strongly recommend to do this change

```ruby
while players.length < 2
  p "Player #{players.length + 1} enter your name: "
  name = gets.chomp
  if !name.empty?
    players << name
  else
    p 'invalid name entered'
  end
end
```

- [ ] Another change @majovanilla mentioned is that when the game is a tie your board is not updating the last move, this is also optional but I also recommend you update the board with all the moves when it's a tie.

image


### That's it, team! :raised_hands: 

### Please :pray: do the changes (or not) :technologist: :technologist: and merge this branch when you are done :memo: 

### Keep up the good work! :clinking_glasses: 

PROJECT STATUS: APPROVED :heavy_check_mark:

**Pull request merged**
